src/Symfony/Component/Validator/Validator/RecursiveContextualValidator.php

```diff
@@ -590,9 +590,7 @@ private function validateClassNode($object, $cacheKey, ClassMetadataInterface $m
                     $object,
                     $cacheKey.':'.$propertyName,
                     $propertyMetadata,
-                    $propertyPath
-                        ? $propertyPath.'.'.$propertyName
-                        : $propertyName,
+                    PropertyPath::append($propertyPath, $propertyName),
```

Comment:
\Symfony\Component\Validator\Tests\Validator\RecursiveValidator2Dot5ApiTest is the one or rather the parent abstract class